### DISCUSSION TITLE
Werden die neuen Hoymiles HMT-4T Wechselrichter unterstützt?

### DISCUSSION THREAD

**Discussion by io2345** [tbnobody/OpenDTU#1469] [Q&A]
Die aktuelle Hoymiles Produktlinie der 3-phasigen WR umfasst auch Modelle mit 4 DC-Eingängen, HMT-1600-4T, 1800 und 2000. Alle haben den Sub-1G-Standard der anderen HMT-Modelle, sind aber (bisher) nicht in der Kompatibilitätsliste aufgeführt. Deshalb die Frage: Werden die Modelle unterstützt?

**Comment by io2345** [comment]
Müsste so sein, da das Handbuch sagt: "_HMT-1600/1800/2000-4T is only compatible with Hoymiles gateway DTU-Pro-S and DTU-Lite-S_"

**Comment by stefan123t** [comment]
HMS/HMT Modelle verwenden alle das HMS101 Funk-Modul von Hoymiles mit einem CMT2300A dh das RF-Modul sollte kompatibel sein. Ich weiß nicht sicher welches Modell-Prefix die von Dir genannten HMT Modelle verwenden. Dh evtl kann es notwendig sein den OpenDTU Code dazu zu bringen auch Seriennummern mit diesem Modell-Prefix zu akzeptieren.
Weiter kann es notwendig sein das Protokoll bzw die Antworten für die og 3 Phasen / 4 DC Channels anzupassen.

Cave at Hoymiles: seit diesem Jahr gibt es auch HMS-…-..W Modelle die über eine built-in WiFi DTUBI verfügen. Diese kommuniziert dann wieder auf 2.4GHz aber über WLAN/WiFi.

**Comment by io2345** [comment]
Prefix sagt mir nichts. Alle Seriennummern beginnen aber mit 1361922..., falls das gemeint ist.

**Comment by io2345** [comment]
Habe mittlerweile eine OpenDTU am Laufen mit diesem Wechselrichtertyp. Bekomme keine Livedaten angezeigt,  bei den eingetragenen Wechselrichter-Seriennummern steht unter "Typ" nur _Unknown_. Nun weiss ich nicht, ob's am Zusammenbau liegt, oder ob der WR nicht mit der OpenDTU spricht. 
Hat noch keiner diese Kombination am Laufen?
Denke aber, dass Typ _Unknown_ das Problem sein dürfte, richtig? 
Dann müssten sich wohl die Entwickler damit auseinandersetzen. Lesen die hier mit, oder wie kann ich die informieren?

**Comment by tbnobody** [comment]
Ich kann dir hier mal eine Testfirmware bauen mit der du versuchen kannst die Rohdaten zu lesen. Du müsstest dann von der Konsole die entsprechenden Ausgaben hier bereitstellen.

Hast du den source selbst kompiliert oder ein fertiges .bin verwendet? Wenn ja, welches?

**Comment by io2345** [comment]
Prima. Die Ausgaben von der Konsole, landen die automatisch in einer TMP-Datei, oder wie bekommt man die am geschicktesten abgespeichert?
Bisher habe ich in der Konsole nur den Countdown der Admin-Schnittstelle zu sehen bekommen

**Comment by tbnobody** [comment]
In einer TMP-Datei landet da nichts. Du gehst einfach auf Info --> Konsole und hältst nach folgendem Ausschau:

```
14:03:18.203 > TX RealTimeRunData Channel: 40 --> 15 71 60 35 46 80 12 23 04 80 0B 00 65 47 92 94 00 00 00 00 00 00 00 00 82 5B B4 
14:03:18.301 > Interrupt received
14:03:18.404 > RX Channel: 3 --> 95 71 60 35 46 71 60 35 46 02 00 0C 57 D7 01 1D 01 12 01 67 01 04 00 02 03 A4 D2 | -80 dBm
14:03:18.459 > Interrupt received
14:03:18.607 > RX Channel: 75 --> 95 71 60 35 46 71 60 35 46 03 00 06 00 0C 7F 92 00 00 11 DB 01 0B 00 03 08 DC 66 | -80 dBm
14:03:18.817 > Interrupt received
14:03:18.868 > RX Channel: 75 --> 95 71 60 35 46 71 60 35 46 84 13 89 0B 65 00 CB 00 81 03 E6 00 B7 00 06 C7 E1 DD | -80 dBm
14:03:19.016 > RX Period End
14:03:19.016 > Middle missing
14:03:19.016 > Request retransmit: 1
14:03:19.016 > TX RequestFrame Channel: 61 --> 15 71 60 35 46 80 12 23 04 81 43 
14:03:19.120 > Interrupt received
14:03:19.222 > RX Channel: 3 --> 95 71 60 35 46 71 60 35 46 01 00 01 01 5B 01 35 01 33 04 2E 04 26 00 0C 5F D3 41 | -80 dBm
14:03:19.325 > RX Period End
14:03:19.325 > Success
```

Wichtig hierbei ist das `TX RealTimeRunData` mit den darauffolgenden `RX` Antworten und dem  `Success` am Ende. Das siehst du aber erst mit der entsprechenden Firmware die ich noch bauen muss. Dafür muss ich aber wissen ob du selbst übersetzt hast oder welche .bin Datei du verwendet hast.

**Comment by io2345** [comment]
Verstehe. Habe die Datei verwendet:
https://github.com/tbnobody/OpenDTU/releases/download/v23.10.9/opendtu-generic.factory.bin
Hatte die Datei am 02.November heruntergeladen - also, wenn sich zwischenzeitlich die Version geändert hat, wäre es u.U. auch die Vorgängerversion

**Comment by tbnobody** [comment]
[firmware.zip](https://github.com/tbnobody/OpenDTU/files/13259811/firmware.zip)
Du solltest diese Firmware nach dem extrahieren über die Web GUI updaten können. In der Inverter Übersicht sollte dann auch kein "Unknown" mehr erscheinen.

**Comment by tbnobody** [comment]
> nach Anleitung eine DTU mit dem CMT2300A-Modul zusammengebaut und mit dem HMS-1600-4T ohne Probleme in Betrieb nehmen können.

Vorsicht: In diesem Thread geht es nicht um mden HMS-1600-4T sondern um den HMT-1600-4T!!

**Reply by sp4m4r** [reply]
Oh sorry, das tut mir sehr leid. Ich bin noch recht neu dabei. Ich lösche meine Antwort.

**Comment by io2345** [comment]
Schaut gut aus (siehe Anhang)
[WR1.txt](https://github.com/tbnobody/OpenDTU/files/13264297/WR1.txt)

Wird aktuell mit 6 DC-Eingängen angezeigt, hat aber nur vier. 
![WR1](https://github.com/tbnobody/OpenDTU/assets/111304560/9284cd02-8138-4f95-8ebb-3d91892be671)

Was mir noch unklar ist: Wie wird das Limit gesetzt. Bedeutet ein Limit von 100%, dass der Wechselrichter mit 100% läuft, oder dass er komplett gedrosselt ist?
EDIT: Schon gefunden: Limit 100% entspricht volle Leistung

**Comment by tbnobody** [comment]
> Wird aktuell mit 6 DC-Eingängen angezeigt, hat aber nur vier.

Ja. Weil ich aktuell mangels Kommunikationsdump (den du mir ja jetzt hochgeladen hast) einfach die Zuordnung des HMT-6CH kopiert habe. Wobei die Werte ansich so schlecht nicht aussehen. Hast du aktuell nur 1 Panel angeschlossen?

> Bedeutet ein Limit von 100%, dass der Wechselrichter mit 100% läuft

Ja

**Comment by io2345** [comment]
Genau, für den Test nur ein Panel dran. Anlage kommt erst in den nächsten Wochen aufs Dach.

**Comment by tbnobody** [comment]
Könntest du ggf. mal testen ob die Kanalzuordnung passt? Also wenn du dein Panel an Kanal 1 ansteckst, ob dann die entsprechenden Werte auch bei Kanal 1 erscheinen bzw. die Zuordnung für die Kanäle 2-4 ebenfalls passt?

**Comment by io2345** [comment]
Gerne. Hab meinen Test zwar mittlerweile wieder abgebaut, aber daran soll's nicht scheitern. Heute natürlich schon zu dunkel, morgen dann.

Edit: Zuordnung passt. Also nur String 5 und 6 entfernen. 
Hab heute auch nochmal ein Log mit einem anderen Modell gemacht (HMT-1800-4T). Häng ich an.
[1800er.txt](https://github.com/tbnobody/OpenDTU/files/13279153/1800er.txt)

**Comment by io2345** [comment]
Hallo, habe gesehen, dass die HMT-xxx-4T Modelle jetzt auf der Kompatibilitätsliste stehen. In dieser Liste müsste noch die Anzahl der MPP-Tracker korrigiert werden. Diese Modelle haben je einen MPP-Tracker für zwei Eingänge, insgesamt also zwei (nicht vier, wie derzeit in der Liste angegeben).
